src/SaleHandler.sol

```diff
@@ -32,24 +48,40 @@ contract SaleHandler {
     }
 
     function onReceive() external {
+        require(repayBps == 10000 || beneficiary != address(0), "Either all goes to repayment or beneficiary must be set");
```

Comment:
I think it's dangerous to have the `onReceive()` revert under any circumstances. It's better to have it either redirect any funds towards repayment, or store the funds in the contract in case of a lack of `beneficiary`. If the auction logic is interrupted, as soon as a change is made to params that allow for a successful auction, a lot of DBR will be able to be bought very cheaply.

Something like the following:
```suggestion
        uint bal = dola.balanceOf(address(this));
        uint repayment = bal * repayBps / 10000;
        if(beneficiary == address(0) && repayBps < 10000) repayment = dola.balanceOf(address(this));
```